torchtitan/parallelisms/pipeline_llama.py

```diff
@@ -141,6 +146,30 @@ def _build_stage(stage_idx, start_layer, stop_layer, is_first=False, is_last=Fal
         )
         return stage, model
 
+    # if split points are not specified, we split the model into equal chunks based on
+    # the number of pipeline stages.
+    if len(splits) == 0:
+        # we assume num_stages per rank based on the schedule time
+        schedule_class = get_schedule_class(
+            job_config.experimental.pipeline_parallel_schedule
+        )
+        if issubclass(schedule_class, PipelineScheduleSingle):
+            num_stages_per_rank = 1
+        elif issubclass(schedule_class, PipelineScheduleMulti):
+            num_stages_per_rank = 2
+        else:
+            raise ValueError(
+                f"Unsupported pipeline schedule: {job_config.experimental.pipeline_parallel_schedule}"
+            )
+        total_stages = parallel_dims.pp * num_stages_per_rank
+        num_layers = model_config.n_layers
+        if total_stages > num_layers:
+            raise ValueError("Total stages cannot be greater than the number of layers")
+        interval = num_layers // total_stages
+        # Generate split points
+        splits = ["layers." + str(i * interval) for i in range(1, total_stages)]
```

Comment:
The goal of this PR is not to perform optimal auto splitting. To do so, we would need full graph capture of the model and a much more sophisticated algorithm (e.g. https://github.com/pytorch/PiPPy/blob/main/pippy/graphsplit.py#L52-L65). 

I think this addition makes sense. Hopefully it doesn't complicate the code more. but I do need to make this generic for all llama sizes. I moved the definition into the `pipelining_utils.py` file. The main goal of the PR is to provide a default so that users don't need to write out `pipeline_parallel_split_points`, but we will still allow them to do so if they want to tune more granularly, so I added a logger message to hint this as well.